response_operations_ui/views/collection_exercise.py

```diff
@@ -37,12 +51,21 @@ def view_collection_exercise(short_name, period, error=None, success_panel=None,
         }
     ]
 
+    success_key = request.args.get('success_key')
+    error_key = request.args.get('error_key')
+    if success_key:
```

Comment:
If you use .get() in get_success_message and get_error_message you won't need this if else statement.
```
success_message = get_success_message(success_key)
error_message = get_error_message(error_key)
```